neo/rawio/neuronexusrawio.py

```diff
@@ -0,0 +1,275 @@
+        if Path(filename).suffix != ".json":
+            raise NeoReadWriteError(
+                f"The json metadata file should be given, filename entered is {Path(filename).stem}"
+            )
+        meta_filename = Path(filename)
+        binary_file = meta_filename.parent / (meta_filename.stem.split(".")[0] + "_data.xdat")
+
+        if not binary_file.exists() and not binary_file.is_file():
+            raise FileNotFoundError(f"The data.xdat file {binary_file} was not found. Is it in the same directory?")
+        timestamp_file = meta_filename.parent / (meta_filename.stem.split(".")[0] + "_timestamp.xdat")
+        if not timestamp_file.exists() and not timestamp_file.is_file():
+            raise FileNotFoundError(
+                f"The timestamps.xdat file {timestamp_file} was not found. Is it in the same directory?"
+            )
+
+        self.filename = filename
+        self.binary_file = binary_file
+        self.timestamp_file = timestamp_file
+
+    def _source_name(self):
+        # return the metadata filename only
+        return self.filename
+
+    def _parse_header(self):
+
+        # read metadata
+        self.metadata = self.read_metadata(self.filename)
+
+        # Collect information necessary for memory map
+        self._sampling_frequency = self.metadata["status"]["samp_freq"]
+        self._n_samples, self._n_channels = self.metadata["status"]["shape"]
+        # Stored as a simple float32 binary file
+        BINARY_DTYPE = "float32"
+        binary_file = self.binary_file
+        timestamp_file = self.timestamp_file
+
+        # Make the two memory maps
+        self._raw_data = np.memmap(
+            binary_file,
+            dtype=BINARY_DTYPE,
+            mode="r",
+            shape=(self._n_samples, self._n_channels),
+            offset=0,  # headerless binary file
+        )
+        self._timestamps = np.memmap(
+            timestamp_file,
+            dtype=np.int64,  # this is from the allego sample reader timestamps are np.int64
+            mode="r",
+            offset=0,  # headerless binary file
+        )
+
+        # We can do a quick timestamp check to make sure it is the correct timestamp data for the
+        # given metadata
+        if self._timestamps[0] != self.metadata["status"]["timestamp_range"][0]:
+            raise NeoReadWriteError(f"The metadata indicates a different starting timestamp")
+
+        # organize the channels
+        signal_channels = []
+        channel_info = self.metadata["sapiens_base"]["biointerface_map"]
+
+        # as per dicussion with the Neo/SpikeInterface teams stream_id will become buffer_id
+        # and because all data is stored in the same buffer stream for the moment all channels
+        # will be in stream_id = 0. In the future this will be split into sub_streams based on
+        # type but for now it will be the end-users responsability for this.
+        stream_id = 0  # hard-coded see note above
```

Comment:
I think this should be string to not add more confusing to the already confusing stream_id vs stream_index. At least type increases the gap.

Reply:
I need to check with @samuelgarcia, I honestly can't remember if he had any typing rules for the streams or not. I thought both indexes and ids were ints for the streams, but maybe that was just me seeing a pattern was there wasn't one.

Reply:
There are definitley some stream_ids that are strings. Plexon that I was just working on for example.

Reply:
Works for me then. Will change when I change names.

Reply:
Fixed.